tests/test_actions.py

```diff
@@ -603,8 +605,24 @@ def test_stops_at_first_failure(self, Tester) -> None:
             ).perform_as(Tester)
 
         assert mock_question.answered_by.call_count == 2
+        assert [r.msg for r in caplog.records] == [
+            "Tester sees if all of 4 tests pass:",
+            "    Tester sees if fakeQuestion is equal to <True>.",
+            "        ... hoping it's equal to <True>.",
+            "            => <True>",
+            "    Tester sees if fakeQuestion is equal to <False>.",
+            "        ... hoping it's equal to <False>.",
+            "            => <False>",
+            # don't be fooled! the next few lines do not have commas on purpose
+            "        ***ERROR***\n"
+            "\n"
+            "AssertionError: \n"
+            "Expected: <False>\n"
+            "     but: was <True>\n",
```

Comment:
I don't think this test belongs in this test class. I think we should put it in `TestSee`.

I do think it's worth it to have a test that each of the Actions which now use `represent_prop` are representing the props how we want them to be represented, but `SeeAllOf` doesn't directly use it.

Reply:
I think this test isn't _only_ confirming that `represent_prop` does the right thing.  It's a confluence of everything that goes into making `SeeAllOf` work as expected. 

- beat messages have to format and indent properly
- settings have to work properly
- See has to function properly
- SeeAllOf has to loop through each of the passed in tests.

Reply:
I think of it like this -- we want to prove `SeeAllOf` calls `See`, we can do that with another test showing how many times it gets called.  But say someone tries to sneak in the following change:
`the_actor.should(Silently(See.the(question, resolution)))`
The test with the number of calls wouldn't catch that whereas _this_ test would.

Reply:
I dunno, i think it's still in the wrong place and is (well, would be—we don't have the test that i'm suggesting in `TestSee`) testing things that are already tested elsewhere. 

I personally tend towards not having tests that are a group of tests we already have unless that integration is testing something new. I don't see anything new being tested with this test, aside from the test that is missing from `TestSee` of course.

Reply:
In principle I agree that unit tests should be as atomic as possible. I've always struggled with testing functionality that basically coalesces multiple functionalities.  Each of the smaller functions have been tested but how to you prove they all came together in the right manner?  I found that at some point you'll get some overlap.

"We aren't proving the sugar is sweet, we are proving we used it in the recipe."

Reply:
I can add the following test to `TestSee`
```python
    def test_logging(self, Tester, caplog) -> None:
        with caplog.at_level(logging.INFO):
            See(SimpleQuestion(), IsEqualTo(True)).perform_as(Tester)

        assert [r.msg for r in caplog.records] == [
            "Tester sees if simpleQuestion is equal to <True>.",
            "    Tester examines SimpleQuestion",
            "        => <True>",
            "    ... hoping it's equal to <True>.",
            "        => <True>"
        ]
```

Reply:
I added two tests to `See` to prove the logging works there as expected.

Reply:
For me, i want to know why a test at this level failed just by reading the title, if possible. 

I think you might like feature tests, which test things _as they would be used_ by a user, rather than all this fiddling with unit tests. So instead of testing each part down here, we'd have a test that was like:

```python
def test_click_does_a_click(self, Tester):
    target = Target.the("test").located_by("//test")

    Tester.attempts_to(Click.on_the(Target))

    # assert the dummy element was asked for by "//test"
    # assert the dummy element had `.click()` called
```

It's a pretty interesting concept that turns the tests into more documentation on how the developers expect the tool to be used. If you remember a few years ago, the tests in here used to be kind of like that, until i realized i could just do `Click().perform(Tester)` instead and skipped a layer.

I think the tests as they are now are easier to organize, but i did kind of like how those older tests looked.

Reply:
I can't find the article now that introduced me to the concept because i specifically remember it going on at length about how neat it is that someone wanting to use your library can just look at your unit tests to see how to use it. :(

Reply:
I think the trick to any test suite like this is no one way will suffice.  We want as comprehensive as coverage as we can get. 

For instance, if we were to completely rewrite `SeeAllOf` to no longer use `See` but we still wanted the logging to look the same, we need a test to show that it works as expected.  This treats the action like a black box.  There may be overlap at a unit test level, but only because we _know_ what's inside that action.

Reply:
>  I want to know why a test at this level failed just by reading the title,

Perhapse I should rename these tests.  Something like `test_logging_output` ?